js/src/orderbook.ts

```diff
@@ -85,20 +85,37 @@ export class Orderbook {
    * @param uiAmount Optional, whether to return the amounts in uiAmount
    * @returns Returns an L2 orderbook
    */
-  getL2(depth: number, asks: boolean, uiAmount?: boolean) {
+  getL2(depth: number, asks: boolean, uiAmount = false) {
     const convert = (p: aaob.Price) => {
       return {
-        price: p.price,
-        size: p.size.divn(Math.pow(10, this.market.baseDecimals)),
+        price: computeUiPrice(this._market, p.price),
+        size: divideBnToNumber(
+          p.size,
+          new BN(10).pow(new BN(this.market.baseDecimals))
+        )
       };
     };
     if (uiAmount) {
-      return asks
-        ? this._slabAsks.getL2DepthJS(depth, asks).map(convert)
-        : this._slabBids.getL2DepthJS(depth, asks).map(convert);
+      return (
+        asks
+          ? this._slabAsks.getL2DepthJS(depth, asks).map(convert)
+          : this._slabBids.getL2DepthJS(depth, asks).map(convert)
+      ) as any;
     }
-    return asks
-      ? this._slabAsks.getL2DepthJS(depth, asks)
-      : this._slabBids.getL2DepthJS(depth, asks);
+    return (
+      asks
+        ? this._slabAsks.getL2DepthJS(depth, asks)
+        : this._slabBids.getL2DepthJS(depth, asks)
+    ) as any;
```

Comment:
Please remove the `as any`

Can do something like this for instance

```js
  getL2(depth: number, asks: boolean, uiAmount = false) {
    const convert = (p: aaob.Price) => {
      return {
        price: computeUiPrice(this._market, p.price),
        size: divideBnToNumber(
          p.size,
          new BN(10).pow(new BN(this.market.baseDecimals))
        )
      };
    };
    if (uiAmount) {
      return (
        asks
          ? this._slabAsks.getL2DepthJS(depth, asks).map(convert)
          : this._slabBids.getL2DepthJS(depth, asks).map(convert)
      ) as UiPrice[];
    }
    return (
      asks
        ? this._slabAsks.getL2DepthJS(depth, asks)
        : this._slabBids.getL2DepthJS(depth, asks)
    ) as aaob.Price[];
  }
}

interface UiPrice {
  price: number;
  size: number;
}
```

Reply:
Fixed so that it doesn't need to use `as` typecasts anymore. Note - needed to create separate declaration and definition as Typescript won't reason about conditional type with a free boolean type parameter.

(See: https://stackoverflow.com/questions/52817922/typescript-return-type-depending-on-parameter)